pkg/agent/agent.go

```diff
@@ -111,41 +97,49 @@ func InitializeWithAgentFeatures(centralCfg config.CentralConfig, agentFeaturesC
 		}
 	}
 
+	// Only create the api map cache if it does not already exist
+	if agent.cacheManager == nil {
+		agent.cacheManager = agentcache.NewAgentCacheManager(centralCfg)
+	}
+
 	if centralCfg.GetUsageReportingConfig().IsOfflineMode() {
 		// Offline mode does not need more initialization
 		agent.cfg = centralCfg
 		return nil
 	}
 
+	agent.cfg = centralCfg
+	coreapi.SetConfigAgent(centralCfg.GetEnvironmentName(), isRunningInDockerContainer(), centralCfg.GetAgentName())
+
 	if agentFeaturesCfg.ConnectionToCentralEnabled() {
 		err = initializeTokenRequester(centralCfg)
 		if err != nil {
 			return err
 		}
-		// Init apic client
-		if agent.apicClient == nil {
-			agent.apicClient = apic.New(centralCfg, agent.tokenRequester)
-			agent.apicClient.AddCache(agent.categoryMap, agent.teamMap)
-		} else {
-			agent.apicClient.SetTokenGetter(agent.tokenRequester)
-			agent.apicClient.OnConfigChange(centralCfg)
+		// Init apic client when the agent starts, and on config change.
+		agent.apicClient = apic.New(centralCfg, agent.tokenRequester)
+		agent.apicClient.AddCache(agent.cacheManager.GetCategoryCache(), agent.teamMap)
```

Comment:
If we add the teamMap to the cache manager then maybe we just send in a ref to the cache manager with the New apic client?

Reply:
Or, maybe better, make adding these caches using the functional options pattern?

Reply:
We can have a separate story to move the team map to cache manger. Once we move the team cache to cache manager, we can remove the AddCache method from service client. The reason I think we added them there was to avoid cyclic references and with agent cache management moved to different package that can be avoided.